cmd/kubeadm/app/cmd/phases/upgrade/node/controlplane.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package node
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/pkg/errors"
+
+	"k8s.io/kubernetes/cmd/kubeadm/app/cmd/options"
+	"k8s.io/kubernetes/cmd/kubeadm/app/cmd/phases/workflow"
+	"k8s.io/kubernetes/cmd/kubeadm/app/phases/upgrade"
+	"k8s.io/kubernetes/cmd/kubeadm/app/util/apiclient"
+)
+
+// NewControlPlane creates a kubeadm workflow phase that implements handling of control-plane upgrade.
+func NewControlPlane() workflow.Phase {
+	phase := workflow.Phase{
+		Name:  "control-plane",
+		Short: "Upgrade the control plane instance deployed on this node, if any",
+		Run:   runControlPlane(),
+		InheritFlags: []string{
+			options.DryRun,
+			options.KubeconfigPath,
+		},
+	}
+	return phase
+}
+
+func runControlPlane() func(c workflow.RunData) error {
+	return func(c workflow.RunData) error {
+		data, ok := c.(Data)
+		if !ok {
+			return errors.New("control-plane phase invoked with an invalid data struct")
+		}
+
+		// if this is not a control-plande node, this phase should not be executed
+		if !data.IsControlPlaneNode() {
+			return nil
```

Comment:
here there should be a message that the phase is skipped:
similar to:
https://github.com/kubernetes/kubernetes/blob/2d81f6649ec22592dd406a66320ebf49954fb254/cmd/kubeadm/app/cmd/phases/init/uploadcerts.go#L54

`fmt.Printf("[upgrade] ] Skipping phase. Not a control plane node")`